### PR TITLE
feat: add client_endpoints_override to bq options

### PR DESCRIPTION
**Pull request by GarrettWu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

b/370077669

### CODE REVIEW

**Review by Genesis929 — APPROVED** [review]

File: bigframes/_config/bigquery_options.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def __init__(
         skip_bq_connection_check: bool = False,
         *,
         ordering_mode: Literal["strict", "partial"] = "strict",
+        client_endpoints_override: dict = {},
```
Comment (Genesis929) [inline bigframes/_config/bigquery_options.py:94]: quick question, why not just endpoints_override.

Comment (GarrettWu) [reply bigframes/_config/bigquery_options.py:94]: Those are bq_xxx_client endpoints.

File: bigframes/_config/bigquery_options.py [hunk 2]
```diff
@@ -317,3 +319,21 @@ def ordering_mode(self) -> Literal["strict", "partial"]:
...
+
+    @property
+    def client_endpoints_override(self) -> dict:
+        """Option that sets the BQ client endpoints addresses directly as a dict. Possible keys are "bqclient", "bqconnectionclient", "bqstoragereadclient"."""
+        return self._client_endpoints_override
+
+    @client_endpoints_override.setter
+    def client_endpoints_override(self, value: dict):
+        warnings.warn(
+            "This is an advanced option that sets the endpoints directly. Don't use if you don't know about it."
```
[the remark this answers is not in the document]
Comment (GarrettWu) [reply bigframes/_config/bigquery_options.py:331]: Cool.

**Comment by GarrettWu** [comment]
failed samples are caused by https://github.com/googleapis/python-bigquery-dataframes/pull/909, unrelated to this PR.

**Pull request merged**